aws/rust-runtime/aws-config/src/json_credentials.rs

```diff
@@ -177,7 +199,7 @@ pub(crate) fn parse_json_credentials(
 
 pub(crate) fn json_parse_loop<'a>(
     input: &'a [u8],
-    mut f: impl FnMut(Cow<'a, str>, Cow<'a, str>),
+    mut f: impl FnMut(Cow<'a, str>, &Token<'a>) -> Result<(), InvalidJsonCredentials>,
```

Comment:
The changes to the JSON parsing to support IMDSv2, SSO, and the differing `credential_process` JSON formats were a bit more awkward than I was hoping. I started with changing this loop function to return all value types since the `credential_process` format has a number in it (`Version`) which spidered out a bit.